src/replay.c

```diff
@@ -103,219 +113,146 @@ void replay_event(Replay *rpy, int type, int key) {
 		printf("The replay is OVER\n");
 }
 
-void replay_write_separator(FILE *file) {
-	fputs(":", file);
+void replay_write_string(SDL_RWops *file, char *str) {
+	SDL_WriteLE16(file, strlen(str));
+	SDL_RWwrite(file, str, 1, strlen(str));
 }
 
-void replay_write_string(FILE *file, char *str) {
-	fputs(str, file);
-	replay_write_separator(file);
-}
+int replay_write_stage_event(ReplayEvent *evt, SDL_RWops *file) {
+	SDL_WriteLE32(file, evt->frame);
+	SDL_WriteU8(file, evt->type);
+	SDL_WriteLE16(file, evt->value);
 
-void replay_write_int(FILE *file, int val) {
-	fprintf(file, "%d", val);
-	replay_write_separator(file);
+	return True;
 }
 
-void replay_write_double(FILE *file, double val) {
-	fprintf(file, "%f", val);
-	replay_write_separator(file);
-}
+int replay_write_stage(ReplayStage *stg, SDL_RWops *file) {
+	int i;
 
-void replay_write_complex(FILE *file, complex val) {
-	replay_write_double(file, creal(val));
-	replay_write_double(file, cimag(val));
+	SDL_WriteLE32(file, stg->stage);
+	SDL_WriteLE32(file, stg->seed);
+	SDL_WriteU8(file, stg->diff);
+	SDL_WriteLE32(file, stg->points);
+	SDL_WriteU8(file, stg->plr_char);
+	SDL_WriteU8(file, stg->plr_shot);
+	SDL_WriteLE64(file, stg->plr_pos_x._int);
+	SDL_WriteLE64(file, stg->plr_pos_y._int);
+	SDL_WriteU8(file, stg->plr_focus);
+	SDL_WriteU8(file, stg->plr_fire);
+	SDL_WriteLE64(file, stg->plr_power._int);
```

Comment:
We can probably just dodge this problem. Store power as an int (internally in the game state too, 105 = 1.05 power), round up the coordinates when the stage starts and store them as ints in the replay. Boom, free portability.